src/main/scala/net/snowflake/spark/snowflake/SnowflakeTelemetry.scala

```diff
@@ -366,6 +386,22 @@ object SnowflakeTelemetry {
     }
     metric
   }
+
+  private[snowflake] def detectSparkLanguage(sparkConf: SparkConf): String = {
+    if (sparkConf.contains("spark.r.command")
+      || sparkConf.contains("spark.r.driver.command")
+      || sparkConf.contains("spark.r.shell.command")) {
+      "R"
+    } else if (sparkConf.contains("spark.pyspark.python")) {
+      "Python"
+    } else {
+      "Scala"
```

Comment:
nit: Is there some way to distinguish between Scala and Java? If not, should this be JVM or some combined label?

Reply:
Scala or Java is from development language perspective. The language detection is at RunTime.  I don't think we can distinguish them without some special Spark configuration. `JVM` sounds more generic. But I prefer to use `Scala` because Scala is well accepted in Spark world.